### PR TITLE
[MI-3804]: Added the test cases for the UI design on MS Teams sync plugin.

### PR DESCRIPTION
**Pull request by arush-vashishtha**
### Summary
This PR consists of the test cases for the UI of following features in RHS on MM,

- Connecting to MS Teams account
- Disconnecting from MS Teams account
- Error while connecting to MS Teams account

### CODE REVIEW

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the App bar is enabled, click and open the RHS.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md]:
```suggestion
1. When the App bar is enabled, click on MS Teams icon and open the RHS.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+**Step 2**
+
+1. When the App bar is disabled, click and open the RHS.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md]:
```suggestion
1. When the App bar is disabled, click on MS Teams icon in channel header and open the RHS.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnected_user_with_linked_channel.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the channels are linked from MM to MS Teams disconnect the user from their MS Teams account from the RHS.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnected_user_with_linked_channel.md]: Write complete steps here.

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/connecting_user.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "RHS changes after connecting the user on MM"
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/connecting_user.md]:
```suggestion
name: "RHS updates after connecting the user on MM"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/connecting_user.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Connect the user to their MS Teams account from RHS.
+
+**Expected**
+
+The user account should get connected to the MS Teams account.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/connecting_user.md]: Expected result does not include the result shown reflected in the name of the test case, fix it.

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnected_user_with_linked_channel.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "RHS changes after disconnecting user from their account on MM"
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnected_user_with_linked_channel.md]:
```suggestion
name: "RHS updates after disconnecting user from their account on MM"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnected_user_with_linked_channel.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the channels are linked from MM to MS Teams disconnect the user from their MS Teams account from the RHS.
+
+**Expected**
+
+The user should be able to see the request message to connect your account with the list of linked channels.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnected_user_with_linked_channel.md]:
```suggestion
The user should be able to see the request message to connect your account with the list of linked channels in the RHS.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Click on the disconnect button in the RHS on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md]:
```suggestion
1. Click on the disconnect button in the RHS of MS Teams sync plugin on MM.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+2. CLick on disconnect button in the `Disconnect Microsoft teams account` modal.
+
+**Step 2**
+
+1. Click on the disconnect button in the RHS on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md]: Same as above

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
+---
+# (Required) Ensure all values are filled up
+name: "RHS changes after disconnecting user from their account on MM"
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md]:
```suggestion
name: "RHS updates after disconnecting user from their account on MM"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+2. CLick on disconnect button in the `Disconnect Microsoft teams account` modal.
+
+**Step 2**
+
+1. Click on the disconnect button in the RHS on MM.
+2. CLick on cancel button in the `Disconnect Microsoft teams account` modal.
+
+**Expected**
+
+The user should be able to view the `Connect your Microsoft Teams account` modal in the RHS with the toast message regarding the disconnection of the user account on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/disconnecting_user_without_linked_channel.md]: This is not a modal, fix it and use proper terminologies

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/error_while_connecting_user.md [hunk 12]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "RHS changes for error in connecting the user"
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/error_while_connecting_user.md]:
```suggestion
name: "RHS updates for error in connecting the user"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md [hunk 13]
```diff
@@ -0,0 +1,48 @@
+---
+# (Required) Ensure all values are filled up
+name: "RHS changes with respect to slash commands"
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md]:
```suggestion
name: "RHS updates with respect to slash commands"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Connect the user to their MS Teams account using the `/msteams connect` slash command.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md]: write the correct slash command

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+2. Click and open the RHS.
+
+**Step 2**
+
+1. Disconnect the user from their MS Teams account using the `/msteams disconnect` slash command.
+2. Click and open the RHS.
+
+**Expected**
+
+The user account on MM should get connected to the account on MS Teams in the RHS on MM.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md]:
```suggestion
The user account on MM should get connected to the account on MS Teams and the RHS should get updated accordingly.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+After step 2, the user should be able to view the `Connect your Microsoft Teams account` modal in the RHS on MM and the account should get disconnected.
```
Comment (AayushChaudhary0001) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/slash_commands.md]: this is not a modal, fix this.

**Review by AayushChaudhary0001 — APPROVED** [review]

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Connecting_user.md
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Connecting_user.md:1]: I think we can change the name of the folder to something like `ui-integration`. if changing, update the test cases as well.

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md [hunk 17]
```diff
@@ -0,0 +1,42 @@
+---
+# (Required) Ensure all values are filled up
+name: "Updated RHS after disconnecting a user from their account on MM"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md]:
```suggestion
name: "Updated RHS after disconnecting a user from MS Teams account on MM"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md [hunk 18]
```diff
@@ -0,0 +1,42 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Click on the `Disconnect button` in the RHS of MS Teams sync plugin, when the channels are linked from MM to MS Teams .
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md]: Do we really need to write `MS Teams sync plugin` here?

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md [hunk 19]
```diff
@@ -0,0 +1,42 @@
...
+2. Click on the `Disconnect button` in `Disconnect Microsoft Teams Account` modal on MM.
+
+**Expected**
+
+The user should be able to see the request message to `connect your account` with the list of linked channels in the RHS.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md]: What about the toast message?

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md [hunk 20]
```diff
@@ -0,0 +1,48 @@
+---
+# (Required) Ensure all values are filled up
+name: "Updated RHS after disconnecting a user from their account on MM"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md]: same

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md (hunk 18 above)
Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md]: if updating, fix everywhere

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Click on the `disconnect button` in the RHS of MS Teams sync plugin on MM.
+2. Click on `disconnect button` in the `Disconnect Microsoft teams account` modal.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md]:
```suggestion
1. Click on the `disconnect button` in the RHS of MS Teams sync plugin on MM.
2. Click on the `Disconnect button` in the `Disconnect Microsoft Teams Account` modal.
```

Comment (ayusht2810) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md]: I think we should try to match the casing as well for keywords mentioned in the markdown

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+
+**Step 2**
+
+1. Click on the `disconnect button` in the RHS of MS Teams sync plugin on MM.
+2. Click on `cancel` button or `x` button in the `Disconnect Microsoft teams account` modal.
+
+**Expected**
+
+The user should be able to view the `connect account` button in the RHS with the toast message regarding the successful disconnection of their MS Teams account.
+After step 2, the `Disconnect Microsoft teams account` modal should close and the existing connection of user should not get affected.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md]:
```suggestion
After step 2, the `Disconnect Microsoft Teams account` modal should close and the existing connection of the user should not be affected.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+2. Click on `disconnect button` in the `Disconnect Microsoft teams account` modal.
+
+**Step 2**
+
+1. Click on the `disconnect button` in the RHS of MS Teams sync plugin on MM.
+2. Click on `cancel` button or `x` button in the `Disconnect Microsoft teams account` modal.
+
+**Expected**
+
+The user should be able to view the `connect account` button in the RHS with the toast message regarding the successful disconnection of their MS Teams account.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md]: The difference between the above test file's expected result and this is not much different. Please change them to differentiate between them.

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Error_while_connecting_user.md [hunk 24]
```diff
@@ -0,0 +1,41 @@
+---
+# (Required) Ensure all values are filled up
+name: "Updated RHS for errors while connecting a user"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Error_while_connecting_user.md]:
```suggestion
name: "Error while connecting a user through RHS"
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Error_while_connecting_user.md [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Any error occured while connecting the user to their MS Teams account.
+
+**Expected**
+
+The user should be able to view the `connect account` button in the RHS with the toast message in RHS regarding the failure of the connection due to an error.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Error_while_connecting_user.md]:
```suggestion
The current RHS state should persist and a toast message should appear in RHS regarding the failure of the connection due to an error.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md [hunk 26]
```diff
@@ -0,0 +1,46 @@
...
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the `App bar` is enabled, click on MS Teams icon and open the RHS.
+
+**Step 2**
+
+1. When the `App bar` is disabled, click on MS Teams icon in channel header and open the RHS.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md]:
```suggestion
1. When the `App bar` is disabled, click on the MS Teams sync icon in the channel header and open the RHS.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md [hunk 27]
```diff
@@ -0,0 +1,46 @@
...
+key: null
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. When the `App bar` is enabled, click on MS Teams icon and open the RHS.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md]: same

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md [hunk 28]
```diff
@@ -0,0 +1,46 @@
...
+
+**Step 2**
+
+1. When the `App bar` is disabled, click on MS Teams icon in channel header and open the RHS.
+
+**Expected**
+
+The user should be able to view the RHS of MS Teams along with the `App bar` in the side on MM.
+After step 2, the user should be able to view the RHS of MS Teams without `App bar` in the side and the MS Teams sync plugin icon in the channel header on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md]:
```suggestion
The user should be able to view the RHS of the plugin along with the `App bar` on the side of MM.
After step 2, the user should be able to view the RHS of the plugin along with the channel header on MM.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md [hunk 29]
```diff
@@ -0,0 +1,46 @@
+---
+# (Required) Ensure all values are filled up
+name: "Validating the RHS design"
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/RHS_design.md]: Is the name correct here, as we are not checking the whole rhs functionality but instead just enabling and disabling the app bar?

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md [hunk 30]
```diff
@@ -0,0 +1,48 @@
...
+created_on: null
+last_updated: null
+case_hashed: null
+steps_hashed: null
+---
+
+**Step 1**
+
+1. Connect the user to their MS Teams account using the `/msteams-sync connect` slash command.
+2. Click on MS Teams icon in App bar or channel header and open the RHS.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md]: update icon name, everywhere

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md [hunk 31]
```diff
@@ -0,0 +1,48 @@
...
+
+**Step 2**
+
+1. Disconnect the user from their MS Teams account using the `/msteams-sync disconnect` slash command.
+2. Click on MS Teams icon in App bar or channel header and open the RHS.
+
+**Expected**
+
+The user account should get connected to their MS Teams account and the same should reflect in the RHS on MM.
+After step 2,the user account should get disconnected from their MS Teams account and the user should be able to view the `connect account` button in the RHS on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md]:
```suggestion
The user account should be connected to their MS Teams account and the same should be reflected in the RHS on MM.
After step 2, the user account should be disconnected from their MS Teams account and the user should be able to view the `Connect account` button in the RHS on MM.
```

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md:1]: Should we write a test case for the case when the rhs was already open and the user connects through slash commands to see the auto update of the rhs while it was opened?

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md (hunk 18 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md]: yes

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md:1]: The test case added for the slash command is generic and covers both the RHS state opened and closed.

Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Slash_commands.md:1]: discussed

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md (hunk 23 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnecting_user_without_linked_channel.md]: discussed

File: data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md (hunk 18 above)
Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/UI-design-1.0/Disconnected_user_with_linked_channel.md]: discussed

File: data/test-cases/plugins/ms-teams-sync/ui-integration/Error_while_connecting_user.md [hunk 32]
```diff
@@ -38,4 +38,4 @@ steps_hashed: null
 
 **Expected**
 
-The user should be able to view the `connect account` button in the RHS with the toast message in RHS regarding the failure of the connection due to an error.
\ No newline at end of file
+The current RHS state should persist and a toast message should appear in the RHS of MS Teams sync plugin regarding the failure of the connection due to an error.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/Error_while_connecting_user.md]:
```suggestion
The current RHS state should persist and a toast message should appear in the RHS of the MS Teams sync plugin regarding the failure of the user connection due to an error.
```

File: data/test-cases/plugins/ms-teams-sync/ui-integration/RHS_design.md [hunk 33]
```diff
@@ -34,13 +34,13 @@ steps_hashed: null
...
-1. When the `App bar` is disabled, click on MS Teams icon in channel header and open the RHS.
+1. When the `App bar` is disabled, click on the MS Teams sync icon in the channel header and open the RHS.
 
 **Expected**
 
-The user should be able to view the RHS of MS Teams along with the `App bar` in the side on MM.
-After step 2, the user should be able to view the RHS of MS Teams without `App bar` in the side and the MS Teams sync plugin icon in the channel header on MM.
\ No newline at end of file
+The user should be able to view the RHS of MS Teams sync plugin along with the `App bar` on the side on MM.
+After step 2, the user should be able to view the RHS of MS Teams sync plugin without the `App bar` on the side on MM.
```
Comment (ayusht2810) [inline data/test-cases/plugins/ms-teams-sync/ui-integration/RHS_design.md]: please check the review comment again related to this

Comment (arush-vashishtha) [reply data/test-cases/plugins/ms-teams-sync/ui-integration/RHS_design.md]: discussed.

**Review by ayusht2810 — APPROVED** [review]

**Pull request merged**